### PR TITLE
Add support for the CUE language & Cue Sheet

### PR DESCRIPTION
**Pull request by zchee**
This PR adds support for the [CUE language](https://cuelang.org).

## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3Acue+package+NOT+nothack
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/cuelang/cue/blob/3bdfa5d10b6bc232241eb84d7a0b7761b806bac1/cue.mod/pkg/github.com/SchemaStore/schemastore/src/schemas/json/github-workflow.cue
        - Sample license(s): [Apache-2.0 License](https://github.com/cuelang/cue/blob/master/LICENSE)
      - https://github.com/cuelang/cue/blob/447606047bdd32f79029a4be5424af3a09ea15f2/doc/tutorial/kubernetes/quick/services/kube.cue
        - Sample license(s): [Apache-2.0 License](https://github.com/cuelang/cue/blob/master/LICENSE)
      - https://github.com/zchee/CueSheet-Sample/blob/9cb7d9acb6f5a09602523cbcd83cdd59ec5e3b0c/example1.cue
        - Sample license(s): [Apache-2.0 License](https://github.com/zchee/CueSheet-Sample/blob/main/LICENCE)
      - https://github.com/zchee/CueSheet-Sample/blob/9cb7d9acb6f5a09602523cbcd83cdd59ec5e3b0c/example2.cue
        - Sample license(s): [Apache-2.0 License](https://github.com/zchee/CueSheet-Sample/blob/main/LICENCE)
  
  - [x] I have included a syntax highlighting grammar:
	  - samples/CUE/kube.cue **(Apache-2.0 License)**: https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fgithub.com%2Fcue-sh%2Fvscode-cue%2Fblob%2F7bd21882dc0e5299271597f18f2740394714cfea%2Fsyntaxes%2Fcue.tmLanguage.json&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Fcuelang%2Fcue%2Fblob%2F447606047bdd32f79029a4be5424af3a09ea15f2%2Fdoc%2Ftutorial%2Fkubernetes%2Fquick%2Fservices%2Fkube.cue&code=
	  - samples/CUE/github-workflow.cue **(Apache-2.0 License)**: https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fgithub.com%2Fcue-sh%2Fvscode-cue%2Fblob%2F7bd21882dc0e5299271597f18f2740394714cfea%2Fsyntaxes%2Fcue.tmLanguage.json&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Fcuelang%2Fcue%2Fblob%2F3bdfa5d10b6bc232241eb84d7a0b7761b806bac1%2Fcue.mod%2Fpkg%2Fgithub.com%2FSchemaStore%2Fschemastore%2Fsrc%2Fschemas%2Fjson%2Fgithub-workflow.cue&code=
	  - samples/Cue Sheet/sample1.cue **(Apache-2.0 License)**: https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fgithub.com%2Frelikd%2FCUE-Sheet_sublime%2Fblob%2F43c66ef78749b2ccf927159a568ed988ddda98dd%2FCUE%2520Sheet.tmLanguage&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Fzchee%2FCueSheet-Sample%2Fblob%2F9cb7d9acb6f5a09602523cbcd83cdd59ec5e3b0c%2Fexample1.cue&code=
		  - https://github.com/zchee/CueSheet-Sample
	  - samples/Cue Sheet/sample2.cue **(Apache-2.0 License)**: https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fgithub.com%2Frelikd%2FCUE-Sheet_sublime%2Fblob%2F43c66ef78749b2ccf927159a568ed988ddda98dd%2FCUE%2520Sheet.tmLanguage&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Fzchee%2FCueSheet-Sample%2Fblob%2F9cb7d9acb6f5a09602523cbcd83cdd59ec5e3b0c%2Fexample2.cue&code=
		  - https://github.com/zchee/CueSheet-Sample
  - [x] I have updated the heuristics to distinguish my language from others using the same extension.

---

### TODO
- [x] Add "Cue Sheet" language support

### CODE REVIEW

**Comment by myitcv** [comment]
@zchee wonderful!

Cross referencing https://github.com/github/linguist/issues/5013

@lildude any pointers/additional things we should consider as part of this PR?

**Comment by lildude** [comment]
> @lildude any pointers/additional things we should consider as part of this PR?

Yes... https://github.com/github/linguist/issues/5013#issuecomment-804263183

**Comment by zchee** [comment]
@lildude ah, What should I do? edit https://github.com/github/linguist/blob/master/lib/linguist/heuristics.yml ...?

**Comment by lildude** [comment]
What we will need is for cue sheet to be added as part of this PR too so this PR will then implement the two most prominent languages that currently use the `.cue` extension on GitHub.

I don't think a heuristic will be needed as the two languages are quite distinct so the classifier should be able to differentiate between the two. Adding two good samples of each would certainly help.

So to put it into points of things to do in this PR:

- Add support for "Cue Sheet" as a new data language
- Add two cue sheet samples
- Add another CUE sample

**Comment by zchee** [comment]
@lildude thanks for very details reply.
And I understand all, will do and mention to you after done.

**Comment by zchee** [comment]
@lildude added Cue Sheet language support. PTAL.

**Comment by zchee** [comment]
@lildude sorry, fixed CI error. PTAL again.

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -1113,6 +1121,13 @@ Cuda:
   codemirror_mime_type: text/x-c++src
   color: "#3A4E3A"
   language_id: 77
+Cue sheet:
```
Comment (lildude) [inline lib/linguist/languages.yml:1124]: Titlecase is preferred: 

```suggestion
Cue Sheet:
```

File: vendor/README.md [hunk 2]
```diff
@@ -100,6 +101,7 @@ This is a list of grammars that Linguist selects to provide syntax highlighting
 - **Csound Document:** [nwhetsell/language-csound](https://github.com/nwhetsell/language-csound)
 - **Csound Score:** [nwhetsell/language-csound](https://github.com/nwhetsell/language-csound)
 - **Cuda:** [harrism/sublimetext-cuda-cpp](https://github.com/harrism/sublimetext-cuda-cpp)
+- **Cue sheet:** [relikd/CUE-Sheet_sublime](https://github.com/relikd/CUE-Sheet_sublime)
```
Comment (lildude) [inline vendor/README.md:104]: As above.

```suggestion
- **Cue Sheet:** [relikd/CUE-Sheet_sublime](https://github.com/relikd/CUE-Sheet_sublime)
```

**Review by lildude — CHANGES_REQUESTED** [review]
Thanks for adding the Cue Sheet support.

Can you please rename it to use titlecase as I've commented inline. You'll need to rename the directory holding the samples too.

Can you please also update the template to link to the source of all four of the samples in this PR and state the license for each file.

🙇

**Comment by zchee** [comment]
@lildude thanks, will fix

**Comment by zchee** [comment]
@lildude PTAL.

**Review by lildude — APPROVED** [review]
😍 Thanks @zchee.

You've just missed the last release so GitHub.com won't gain support until the next release is made which will be in about a month's time.

**Comment by zchee** [comment]
@lildude Thanks a lot!

**Pull request merged**